### DISCUSSION TITLE
Can not understand alternative of deprecated method in `SharedInformerFactory`

### DISCUSSION THREAD

**Discussion by pan3793** [fabric8io/kubernetes-client#5008] [Q&A] [answered]
Since `SharedInformerFactory#inNamespace` is marked as `@deprecated`, and the suggested alternative is `use {@link Informable} instead`, but, it's confusing how to migrate a "method" to an "interface"?

```
public interface SharedInformerFactory {

  /**
   * Configure Namespace for {@link SharedInformerFactory}
   *
   * @param namespace namespace to configure
   * @return {@link SharedInformerFactory} with namespace configured
   * @deprecated use {@link Informable} instead
   */
  @Deprecated
  SharedInformerFactory inNamespace(String namespace);

  /**
   * Configure Name for {@link SharedInformerFactory}
   *
   * @param name name to be configured
   * @return {@link SharedInformerFactory} with name configured
   * @deprecated use {@link Informable} instead
   */
  @Deprecated
  SharedInformerFactory withName(String name);

  ...
}
```

**Comment by pan3793** [comment, marked as answer]
Let's say I have the following code, how should I do the migration?
```
client.informers().inNamespace("testing").sharedIndexInformerFor(Pod.class, 3000)
```

**Reply by rohanKanojia** [reply]
I think you should do `client.pods().inNamespace("testing").inform(..)`

**Reply by pan3793** [reply]
@rohanKanojia thanks, it would be nice if the lib can add more details on the `@deprecated` information.

**Reply by rohanKanojia** [reply]
@pan3793 : Would you like to submit a PR to improve the existing javadoc related to the current method?

**Reply by pan3793** [reply]
Would like to, but maybe after I have more understanding about the library.
